--- src/Terraria/Terraria/IngameOptions.cs
+++ src/Terraclient/Terraria/IngameOptions.cs
@@ -18,7 +_,8 @@
 	{
 		public const int width = 670;
 		public const int height = 480;
-		public static float[] leftScale = new float[10] {
+		public static float[] leftScale = new float[11] {
+			0.7f,
 			0.7f,
 			0.7f,
 			0.7f,
@@ -176,7 +_,7 @@
 			int num7 = flag5 ? 1 : 0;
 			int num8 = 5 + num7 + 2;
 			Vector2 vector = new Vector2(value3.X + value2.X / 4f, value3.Y + (float)(num3 * 5 / 2));
-			Vector2 vector2 = new Vector2(0f, value2.Y - (float)(num3 * 5)) / (num8 + 1);
+			Vector2 vector2 = new Vector2(0f, value2.Y - (float)(num3 * 7.5f)) / (num8 + 1);
 			if (flag)
 				vector.X -= 55f;
 
@@ -242,6 +_,16 @@
 				if (flag4) {
 					Close();
 					IngameFancyUI.OpenKeybinds();
+				}
+			}
+
+			num10++;
+			if (DrawLeftSide(sb, Language.GetTextValue("UI.Cheats"), num10, vector, vector2, leftScale)) {
+				leftHover = num10;
+
+				if (flag4) {
+					Close();
+					IngameFancyUI.OpenCheats();
 				}
 			}
 
