--- src/Terraria/Terraria/Main.cs
+++ src/Terraclient/Terraria/Main.cs
@@ -68,6 +_,10 @@
 using Terraria.Net;
 using Terraria.ObjectData;
 using Terraria.Social;
+using Terraria.Terraclient;
+using Terraria.Terraclient.Cheats;
+using Terraria.Terraclient.Cheats.General;
+using Terraria.Terraclient.GameContent.UI.States;
 using Terraria.UI;
 using Terraria.UI.Chat;
 using Terraria.UI.Gamepad;
@@ -238,7 +_,7 @@
 		public static NPCHeadRenderer BossNPCHeadRenderer = null;
 		public static string versionNumber = "v1.4.1.2";
 		public static string versionNumber2 = "v1.4.1.2";
-		public static string SavePath = Program.LaunchParameters.ContainsKey("-savedirectory") ? Program.LaunchParameters["-savedirectory"] : Platform.Get<IPathService>().GetStoragePath("Terraria");
+		public static string SavePath = (Program.LaunchParameters.ContainsKey("-savedirectory") ? Program.LaunchParameters["-savedirectory"] : Platform.Get<IPathService>().GetStoragePath("Terraria")) + Path.DirectorySeparatorChar + "Terraclient";
 		public static bool AnnouncementBoxDisabled;
 		public static int AnnouncementBoxRange = -1;
 		public static string AutogenSeedName;
@@ -3044,6 +_,8 @@
 
 		public static bool SaveSettings() {
 			Configuration.Clear();
+			foreach (Cheat cheat in CheatHandler.cheats)
+				Configuration.Put(cheat.GetType().Name, cheat.isEnabled);
 			Configuration.Put("ResetDefaultUIScale", false);
 			Configuration.Put("SmartCursorToggle", cSmartCursorModeIsToggleAndNotHold);
 			Configuration.Put("MapEnabled", mapEnabled);
@@ -3158,6 +_,8 @@
 			}
 
 			Configuration.Load();
+			foreach (Cheat cheat in CheatHandler.cheats)
+				Configuration.Get(cheat.GetType().Name, ref cheat.isEnabled);
 			int currentValue = 0;
 			Configuration.Get("LastLaunchedVersion", ref currentValue);
 			bool flag = false;
@@ -8220,6 +_,7 @@
 			});
 
 			base.Window.AllowUserResizing = true;
+			CheatHandler.InitializeCheats();
 			LoadSettings();
 			SetDisplayMonitor();
 			if (screenWidth > GraphicsAdapter.DefaultAdapter.CurrentDisplayMode.Width)
@@ -13017,7 +_,7 @@
 			CurrentFrameFlags.SleepingPlayersCount = num2;
 			if (netMode != 2) {
 				int num3 = myPlayer;
-				if (player[num3].creativeGodMode) {
+				if (player[num3].creativeGodMode || CheatHandler.GetCheat<GodModeCheat>().isEnabled) {
 					player[num3].statLife = player[num3].statLifeMax2;
 					player[num3].statMana = player[num3].statManaMax2;
 					player[num3].breath = player[num3].breathMax;
@@ -15568,7 +_,7 @@
 				numLines++;
 			}
 
-			if ((item.tooltipContext == 0 || item.tooltipContext == 2 || item.tooltipContext == 1 || item.tooltipContext == 3 || item.tooltipContext == 4 || item.tooltipContext == 15 || item.tooltipContext == 6) && LocalPlayer.difficulty == 3 && CreativeItemSacrificesCatalog.Instance.TryGetSacrificeCountCapToUnlockInfiniteItems(item.type, out int amountNeeded)) {
+			if ((item.tooltipContext == 0 || item.tooltipContext == 2 || item.tooltipContext == 1 || item.tooltipContext == 3 || item.tooltipContext == 4 || item.tooltipContext == 15 || item.tooltipContext == 6) && (LocalPlayer.difficulty == 3 || CheatHandler.GetCheat<JourneyModeCheat>().isEnabled) && CreativeItemSacrificesCatalog.Instance.TryGetSacrificeCountCapToUnlockInfiniteItems(item.type, out int amountNeeded)) {
 				int sacrificeCount = LocalPlayerCreativeTracker.ItemSacrifices.GetSacrificeCount(item.type);
 				if (amountNeeded - sacrificeCount > 0) {
 					toolTipLine[numLines] = Language.GetTextValue("CommonItemTooltip.CreativeSacrificeNeeded", amountNeeded - sacrificeCount);
@@ -37227,8 +_,8 @@
 				else if (menuMode == 11) {
 					num2 = 210;
 					num4 = 37;
-					num5 = 8;
+					num5 = 9;
-					array4[num5 - 1] = 8;
+					array4[num5 - 1] = 9;
 					for (int num19 = 0; num19 < num5; num19++) {
 						array7[num19] = 0.75f;
 					}
@@ -37276,6 +_,14 @@
 					}
 
 					num20++;
+					array9[num20] = Language.GetTextValue("UI.Cheats");
+					if (selectedMenu == num20) {
+						SoundEngine.PlaySound(10);
+						MenuUI.SetState(new UIManageCheats());
+						menuMode = 888;
+					}
+
+					num20++;
 					array9[num20] = Lang.menu[103].Value;
 					if (selectedMenu == num20) {
 						SoundEngine.PlaySound(10);
@@ -42780,6 +_,11 @@
 				}
 
 				spriteBatch.Draw(TextureAssets.MapIcon[num138].Value, new Vector2(num136, num137), new Microsoft.Xna.Framework.Rectangle(0, 0, TextureAssets.MapIcon[num138].Width(), TextureAssets.MapIcon[num138].Height()), new Microsoft.Xna.Framework.Color(num139, num139, num139, num139), 0f, default(Vector2), 1f, SpriteEffects.None, 0f);
+
+				// Map teleportation cheat
+				if (CheatHandler.GetCheat<MapTeleportCheat>().isEnabled && mouseRight)
+					player[myPlayer].Teleport((mapFullscreenPos + new Vector2(mouseX - (screenWidth / 2), mouseY - (screenHeight / 2)) / 16 * 16 / mapFullscreenScale) * 16, TeleportationStyleID.DebugTeleport);
+
 				DrawCursor(DrawThickCursor());
 			}
 
