--- src/Terraria/Terraria/Main.cs
+++ src/Terraclient/Terraria/Main.cs
@@ -71,6 +_,11 @@
 using Terraria.Net;
 using Terraria.ObjectData;
 using Terraria.Social;
+using Terraria.Terraclient;
+using Terraria.Terraclient.Cheats;
+using Terraria.Terraclient.Cheats.General;
+using Terraria.Terraclient.Commands;
+using Terraria.Terraclient.GameContent.UI.States;
 using Terraria.UI;
 using Terraria.UI.Chat;
 using Terraria.UI.Gamepad;
@@ -83,9 +_,9 @@
 namespace Terraria
 {
 #if SERVER
-	public class Main : Terraria.Server.Game
+	public partial class Main : Terraria.Server.Game
 #else
-	public class Main : Game
+	public partial class Main : Game
 #endif
 	{
 		public delegate void OnPlayerSelected(PlayerFileData player);
@@ -241,7 +_,7 @@
 		public static NPCHeadRenderer BossNPCHeadRenderer = null;
 		public static string versionNumber = "v1.4.2.2";
 		public static string versionNumber2 = "v1.4.2.2";
-		public static string SavePath = Program.LaunchParameters.ContainsKey("-savedirectory") ? Program.LaunchParameters["-savedirectory"] : Platform.Get<IPathService>().GetStoragePath("Terraria");
+		public static string SavePath = (Program.LaunchParameters.ContainsKey("-savedirectory") ? Program.LaunchParameters["-savedirectory"] : Platform.Get<IPathService>().GetStoragePath("Terraria")) + Path.DirectorySeparatorChar + "Terraclient";
 		public static bool AnnouncementBoxDisabled;
 		public static int AnnouncementBoxRange = -1;
 		public static string AutogenSeedName;
@@ -665,7 +_,7 @@
 		public static int DiscoR = 255;
 		public static int DiscoB;
 		public static int DiscoG;
-		public static int teamCooldown;
+		public static int teamCooldown { get => 0; set => value = 0; }
 		public static int teamCooldownLen = 300;
 		public static bool gamePaused;
 		public static bool gameInactive;
@@ -3055,6 +_,8 @@
 
 		public static bool SaveSettings() {
 			Configuration.Clear();
+			foreach (Cheat cheat in CheatHandler.cheats)
+				Configuration.Put(cheat.GetType().Name, cheat.isEnabled);
 			Configuration.Put("ResetDefaultUIScale", false);
 			Configuration.Put("SmartCursorToggle", cSmartCursorModeIsToggleAndNotHold);
 			Configuration.Put("MapEnabled", mapEnabled);
@@ -3171,6 +_,8 @@
 			}
 
 			Configuration.Load();
+			foreach (Cheat cheat in CheatHandler.cheats)
+				Configuration.Get(cheat.GetType().Name, ref cheat.isEnabled);
 			int currentValue = 0;
 			Configuration.Get("LastLaunchedVersion", ref currentValue);
 			bool flag = false;
@@ -8238,6 +_,7 @@
 			});
 
 			base.Window.AllowUserResizing = true;
+			CheatHandler.InitializeCheats();
 			LoadSettings();
 			SetDisplayMonitor();
 			if (screenWidth > GraphicsAdapter.DefaultAdapter.CurrentDisplayMode.Width)
@@ -13043,7 +_,7 @@
 			CurrentFrameFlags.SleepingPlayersCount = num2;
 			if (netMode != 2) {
 				int num3 = myPlayer;
-				if (player[num3].creativeGodMode) {
+				if (player[num3].creativeGodMode || CheatHandler.GetCheat<GodModeCheat>().isEnabled) {
 					player[num3].statLife = player[num3].statLifeMax2;
 					player[num3].statMana = player[num3].statManaMax2;
 					player[num3].breath = player[num3].breathMax;
@@ -13401,7 +_,7 @@
 			if (!inputTextEnter || !chatRelease)
 				return;
 
-			if (chatText != "") {
+			if (chatText != "" && !CheatCommandHandler.ParseCheatCommand(chatText)) {
 				ChatMessage message = ChatManager.Commands.CreateOutgoingMessage(chatText);
 				if (netMode == 1)
 					ChatHelper.SendChatMessageFromClient(message);
@@ -13422,6 +_,7 @@
 			UILinkPointNavigator.Update();
 			PlayerInput.CacheMousePositionForZoom();
 			PlayerInput.SetZoom_MouseInWorld();
+			HandleClientInput();
 			oldKeyState = keyState;
 			keyState = Keyboard.GetState();
 		}
@@ -15593,7 +_,7 @@
 				numLines++;
 			}
 
-			if ((item.tooltipContext == 0 || item.tooltipContext == 2 || item.tooltipContext == 1 || item.tooltipContext == 3 || item.tooltipContext == 4 || item.tooltipContext == 15 || item.tooltipContext == 6) && LocalPlayer.difficulty == 3 && CreativeItemSacrificesCatalog.Instance.TryGetSacrificeCountCapToUnlockInfiniteItems(item.type, out int amountNeeded)) {
+			if ((item.tooltipContext == 0 || item.tooltipContext == 2 || item.tooltipContext == 1 || item.tooltipContext == 3 || item.tooltipContext == 4 || item.tooltipContext == 15 || item.tooltipContext == 6) && (LocalPlayer.difficulty == 3 || CheatHandler.GetCheat<JourneyModeCheat>().isEnabled) && CreativeItemSacrificesCatalog.Instance.TryGetSacrificeCountCapToUnlockInfiniteItems(item.type, out int amountNeeded)) {
 				int sacrificeCount = LocalPlayerCreativeTracker.ItemSacrifices.GetSacrificeCount(item.type);
 				if (amountNeeded - sacrificeCount > 0) {
 					toolTipLine[numLines] = Language.GetTextValue("CommonItemTooltip.CreativeSacrificeNeeded", amountNeeded - sacrificeCount);
@@ -29758,6 +_,7 @@
 					list.Add(new TextSnippet("|", Microsoft.Xna.Framework.Color.White));
 
 				array = list.ToArray();
+				ChatManager.DrawColorCodedStringWithShadow(spriteBatch, FontAssets.MouseText.Value, ".help", new Vector2(88f, screenHeight - 30), Color.Gray, 0f, Vector2.Zero, Vector2.One);
 				ChatManager.DrawColorCodedStringWithShadow(spriteBatch, FontAssets.MouseText.Value, array, new Vector2(88f, screenHeight - 30), 0f, Vector2.Zero, Vector2.One, out hoveredSnippet);
 				if (hoveredSnippet > -1) {
 					array[hoveredSnippet].OnHover();
@@ -37219,8 +_,8 @@
 				else if (menuMode == 11) {
 					num2 = 210;
 					num4 = 37;
-					num5 = 8;
+					num5 = 9;
-					array4[num5 - 1] = 8;
+					array4[num5 - 1] = 9;
 					for (int num19 = 0; num19 < num5; num19++) {
 						array7[num19] = 0.75f;
 					}
@@ -37268,6 +_,14 @@
 					}
 
 					num20++;
+					array9[num20] = Language.GetTextValue("UI.Cheats");
+					if (selectedMenu == num20) {
+						SoundEngine.PlaySound(10);
+						MenuUI.SetState(new UIManageCheats());
+						menuMode = 888;
+					}
+
+					num20++;
 					array9[num20] = Lang.menu[103].Value;
 					if (selectedMenu == num20) {
 						SoundEngine.PlaySound(10);
@@ -39042,6 +_,7 @@
 			}
 
 			if (!WorldGen.drunkWorldGen) {
+				goto SkipVanillaVersionDrawVers;
 				for (int num107 = 0; num107 < 5; num107++) {
 					Microsoft.Xna.Framework.Color color12 = Microsoft.Xna.Framework.Color.Black;
 					if (num107 == 4) {
@@ -39071,6 +_,64 @@
 					origin2.Y *= 0.5f;
 					spriteBatch.DrawString(FontAssets.MouseText.Value, versionNumber, new Vector2(origin2.X + (float)num108 + 10f, (float)screenHeight - origin2.Y + (float)num109 - 2f), color12, 0f, origin2, 1f, SpriteEffects.None, 0f);
 				}
+
+			SkipVanillaVersionDrawVers:
+				Rectangle mouse = new Rectangle(mouseX, mouseY, 1, 1);
+				const float textOffsetY = 28f;
+				Vector2 constBaseOffset = new Vector2(10, 10);
+
+				void HoverableRect(string text, Vector2 pos, Action whileHovering = null, Action whileNotHovering = null) {
+					Vector2 size = FontAssets.MouseText.Value.MeasureString(text);
+
+					if (new Rectangle((int)pos.X, (int)pos.Y, (int)size.X, (int)size.Y).Contains(mouse))
+						whileHovering?.Invoke();
+					else
+						whileNotHovering?.Invoke();
+				}
+
+				void DrawText(string text, Vector2 pos, Color color) =>
+					ChatManager.DrawColorCodedStringWithShadow(spriteBatch, FontAssets.MouseText.Value, text, pos, color, 0f, Vector2.Zero, Vector2.One);
+
+				string versString = $"Terraclient v{CheatHandler.TCVersion}\nTerraria {versionNumber}";
+
+				DrawText(versString, new Vector2(10f, screenHeight - (textOffsetY * 2f)), Color.White);
+
+				string uuidGen = Language.GetTextValue("Cheats.GenerateUUID");
+				Color generateColor = Color.White;
+				HoverableRect(uuidGen, constBaseOffset, () => {
+					                                        generateColor = Color.Goldenrod;
+					                                        if (mouseLeft && mouseLeftRelease)
+						                                        clientUUID = Guid.NewGuid().ToString();
+				                                        });
+
+				DrawText(uuidGen, constBaseOffset, generateColor);
+				DrawText(Language.GetTextValue("Cheats.CurrentUUID", clientUUID), constBaseOffset + new Vector2(0, textOffsetY), Color.Gray);
+				DrawText(Language.GetTextValue("UI.Computing", Environment.Is64BitProcess ? "64bit" : "32bit"), constBaseOffset + new Vector2(0, textOffsetY * 2f), Color.LightGray);
+				DrawText(Language.GetTextValue("UI.RunningFNA"), constBaseOffset + new Vector2(0, textOffsetY * 3f), Color.LightGray);
+
+				string discordText = Language.GetTextValue("UI.Discord");
+				Vector2 discordTextPos = new Vector2(constBaseOffset.X, constBaseOffset.Y + (textOffsetY * 5f));
+				Color discordColor = Colors.RarityBlue;
+
+				Action hoverDiscord = () => {
+					                      discordColor *= 0.75f;
+					                      discordText += " :D";
+
+					                      string discordLink = "https://discord.com/invite/qrZ4Bpz";
+
+										  DrawText($"({discordLink})", constBaseOffset + new Vector2(8f, textOffsetY * 4f) + FontAssets.MouseText.Value.MeasureString(discordText), Color.White);
+
+					                      if (mouseLeft && mouseLeftRelease) {
+						                      ProcessStartInfo info =
+							                      new(discordLink) { UseShellExecute = true, Verb = "open" };
+						                      Process.Start(info);
+					                      }
+				                      };
+
+				Action notHoverDiscord = () => discordText += " <-";
+
+				HoverableRect(discordText, discordTextPos, hoverDiscord, notHoverDiscord);
+				DrawText(discordText, discordTextPos, discordColor);
 			}
 
 			spriteBatch.End();
@@ -39208,12 +_,19 @@
 		private void PreDrawMenu(out Microsoft.Xna.Framework.Point screenSizeCache, out Microsoft.Xna.Framework.Point screenSizeCacheAfterScaling) {
 			float uiScaleWanted = _uiScaleWanted;
 			float num = (float)screenHeight / 900f;
+			float preScale = num;
 			if (num < 1f)
 				num = 1f;
 
 			if (SettingDontScaleMainMenuUp)
 				num = 1f;
 
+			if (CheatHandler.GetCheat<NoMainMenuUIScalingCheat>().isEnabled)
+				num = 1f;
+
+			if (MenuUI.CurrentState is UIManageCheats uiManageCheats && uiManageCheats.uiScaleOnCtor != num)
+				MenuUI.SetState(new UIManageCheats());
+
 			screenSizeCache = ScreenSize;
 			UIScale = num;
 			PlayerInput.SetZoom_UI();
@@ -42817,6 +_,11 @@
 				}
 
 				spriteBatch.Draw(TextureAssets.MapIcon[num138].Value, new Vector2(num136, num137), new Microsoft.Xna.Framework.Rectangle(0, 0, TextureAssets.MapIcon[num138].Width(), TextureAssets.MapIcon[num138].Height()), new Microsoft.Xna.Framework.Color(num139, num139, num139, num139), 0f, default(Vector2), 1f, SpriteEffects.None, 0f);
+
+				// Map teleportation cheat
+				if (CheatHandler.GetCheat<MapTeleportCheat>().isEnabled && mouseRight)
+					player[myPlayer].Teleport((mapFullscreenPos + new Vector2(mouseX - (screenWidth / 2), mouseY - (screenHeight / 2)) / 16 * 16 / mapFullscreenScale) * 16, TeleportationStyleID.DebugTeleport);
+
 				DrawCursor(DrawThickCursor());
 			}
 
@@ -47201,7 +_,7 @@
 			base.Draw(gameTime);
 			float val = (float)screenWidth / MinimumZoomComparerX;
 			float val2 = (float)screenHeight / MinimumZoomComparerY;
-			ForcedMinimumZoom = Math.Max(Math.Max(1f, val), val2);
+			ForcedMinimumZoom = Math.Max(Math.Max(1f, 1f), 1f);
 			GameViewMatrix.Effects = ((!gameMenu && player[myPlayer].gravDir != 1f) ? SpriteEffects.FlipVertically : SpriteEffects.None);
 			BackgroundViewMatrix.Effects = GameViewMatrix.Effects;
 			BackgroundViewMatrix.Zoom = new Vector2(ForcedMinimumZoom);
