--- src/Terraria/Terraria/Player.cs
+++ src/Terraclient/Terraria/Player.cs
@@ -29,6 +_,7 @@
 using Terraria.Localization;
 using Terraria.ObjectData;
 using Terraria.Social;
+using Terraria.Terraclient;
 using Terraria.UI;
 using Terraria.UI.Chat;
 using Terraria.UI.Gamepad;
@@ -2341,7 +_,7 @@
 		}
 
 		public void ToggleCreativeMenu() {
-			if (dead || difficulty != 3)
+			if (dead || (difficulty != 3 && !CheatHandler.JourneyMode.IsEnabled()))
 				return;
 
 			bool flag = false;
@@ -16137,6 +_,9 @@
 				else
 					Dust.NewDust(new Vector2(position.X + (float)(12 * direction), position.Y + (float)num4 + 4f * gravDir), width - 8, 8, 34, 0f, 0f, 0, default(Color), 1.2f);
 			}
+
+			if (CheatHandler.GodMode.IsEnabled())
+				breath = breathMax;
 		}
 
 		public void CheckCrackedBrickBreak() {
@@ -21160,7 +_,7 @@
 		}
 
 		public void Update_NPCCollision() {
-			if (creativeGodMode)
+			if (creativeGodMode || CheatHandler.GodMode.IsEnabled())
 				return;
 
 			Rectangle rectangle = new Rectangle((int)position.X, (int)position.Y, width, height);
@@ -26472,7 +_,7 @@
 				immuneAlpha = 255;
 
 			UpdateGraveyard(now: true);
-			if (whoAmI == Main.myPlayer && context == PlayerSpawnContext.ReviveFromDeath && difficulty == 3)
+			if (whoAmI == Main.myPlayer && context == PlayerSpawnContext.ReviveFromDeath && (difficulty == 3 || CheatHandler.JourneyMode.IsEnabled()))
 				AutoFinchStaff();
 		}
 
@@ -26677,7 +_,7 @@
 		}
 
 		public double Hurt(PlayerDeathReason damageSource, int Damage, int hitDirection, bool pvp = false, bool quiet = false, bool Crit = false, int cooldownCounter = -1) {
-			if (creativeGodMode)
+			if (creativeGodMode || CheatHandler.GodMode.IsEnabled())
 				return 0.0;
 
 			bool flag = !immune;
@@ -27062,7 +_,7 @@
 		}
 
 		public void KillMe(PlayerDeathReason damageSource, double dmg, int hitDirection, bool pvp = false) {
-			if (creativeGodMode || dead)
+			if (creativeGodMode || dead || CheatHandler.GodMode.IsEnabled())
 				return;
 
 			StopVanityActions();
