--- src/Terraria/Terraria/Projectile.cs
+++ src/Terraclient/Terraria/Projectile.cs
@@ -20,6 +_,8 @@
 using Terraria.ID;
 using Terraria.Localization;
 using Terraria.Physics;
+using Terraria.Terraclient;
+using Terraria.Terraclient.Cheats.General;
 using Terraria.Utilities;
 using Terraria.WorldBuilding;
 
@@ -8680,7 +_,7 @@
 		}
 
 		public void StatusPlayer(int i) {
-			if (Main.player[i].creativeGodMode)
+			if (Main.player[i].creativeGodMode || CheatHandler.GetCheat<GodModeCheat>().IsEnabled)
 				return;
 
 			if (type == 472)
@@ -21430,7 +_,7 @@
 						if (Main.expertMode)
 							timeToAdd = 960;
 
-						if (!Main.player[num724].creativeGodMode)
+						if (!Main.player[num724].creativeGodMode && !CheatHandler.GetCheat<GodModeCheat>().IsEnabled)
 							Main.player[num724].AddBuff(145, timeToAdd);
 					}
 				}
@@ -46464,7 +_,7 @@
 
 				if (Main.netMode != 2) {
 					Player player = Main.player[Main.myPlayer];
-					if (!player.dead && player.active && (player.Center - base.Center).Length() < 300f && !player.creativeGodMode)
+					if (!player.dead && player.active && (player.Center - base.Center).Length() < 300f && !player.creativeGodMode && !CheatHandler.GetCheat<GodModeCheat>().IsEnabled)
 						player.AddBuff(197, 900, quiet: false);
 				}
 			}
