--- src/Terraria/Terraria/Terraria.csproj
+++ src/Terraclient/Terraria/Terraria.csproj
@@ -63,10 +_,13 @@
     <EmbeddedResource Remove="Libraries/Mac/**" Condition="!$(DefineConstants.Contains('MAC'))" />
     <EmbeddedResource Remove="Libraries/Linux/**" Condition="!$(DefineConstants.Contains('LINUX'))" />
     <EmbeddedResource Remove="Libraries/Mono/**" Condition="!$(DefineConstants.Contains('MONO'))" />
-    <EmbeddedResource Remove="Libraries/XNA/**"/>
+    <EmbeddedResource Remove="Libraries/XNA/**" />
-    <EmbeddedResource Remove="Libraries/FNA/**"/>
+    <EmbeddedResource Remove="Libraries/FNA/**" />
     <EmbeddedResource Include="Localization/Content/**" />
     <EmbeddedResource Include="Microsoft/**" />
+  </ItemGroup>
+  <ItemGroup>
+    <None Remove="Localization\Content\en-US\Terraclient.json" />
   </ItemGroup>
   <Target Name="EditBin" AfterTargets="Build">
     <PropertyGroup>
