--- src/tModLoader/Terraria/Terraria.csproj
+++ src/Terraclient/Terraria/Terraria.csproj
@@ -7,7 +_,7 @@
     <Company>Re-Logic</Company>
     <Copyright>Copyright © 2021 Re-Logic</Copyright>
     <RootNamespace>Terraria</RootNamespace>
-    <AssemblyName>tModLoader</AssemblyName>
+    <AssemblyName>Terraclient</AssemblyName>
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
     <UseAppHost>false</UseAppHost>
   </PropertyGroup>
