--- src/tModLoader/Terraria/GameContent/UI/Elements/UIAchievementListItem.cs
+++ src/Terraclient/Terraria/GameContent/UI/Elements/UIAchievementListItem.cs
@@ -3,7 +_,10 @@
 using ReLogic.Content;
 using System;
 using Terraria.Achievements;
+using Terraria.Audio;
+using Terraria.ID;
 using Terraria.Localization;
+using Terraria.Terraclient.Implementation.Achievements;
 using Terraria.UI;
 using Terraria.UI.Chat;
 
@@ -128,6 +_,14 @@
 				vector2.X -= num5 * 1.4f + stringSize2.X;
 				ChatManager.DrawColorCodedStringWithShadow(spriteBatch, FontAssets.ItemStack.Value, text2, vector2, value3, 0f, new Vector2(0f, 0f), baseScale3, 90f);
 			}
+
+			if (_achievement.IsCompleted || !IsMouseHovering)
+				return;
+
+			Main.instance.MouseText(Language.GetTextValue("Terraclient.ClickToUnlock"));
+
+			if (Main.mouseLeft && Main.mouseLeftRelease)
+				_achievement.ForceUnlock();
 		}
 
 		private void UpdateIconFrame() {
@@ -138,6 +_,8 @@
 
 			if (_achievementIcon != null)
 				_achievementIcon.SetFrame(_iconFrame);
+
+			SoundEngine.PlaySound(SoundID.MenuTick);
 		}
 
 		private void DrawPanelTop(SpriteBatch spriteBatch, Vector2 position, float width, Color color) {
@@ -162,6 +_,8 @@
 			base.MouseOut(evt);
 			BackgroundColor = new Color(26, 40, 89) * 0.8f;
 			BorderColor = new Color(13, 20, 44) * 0.8f;
+			SoundEngine.PlaySound(SoundID.MenuTick);
+			Main.instance.MouseText("");
 		}
 
 		public Achievement GetAchievement() => _achievement;
