--- src/Terraria/Terraria/GameContent/UI/Elements/UIWorldListItem.cs
+++ src/Terraclient/Terraria/GameContent/UI/Elements/UIWorldListItem.cs
@@ -8,6 +_,8 @@
 using Terraria.IO;
 using Terraria.Localization;
 using Terraria.Social;
+using Terraria.Terraclient;
+using Terraria.Terraclient.Cheats.General;
 using Terraria.UI;
 
 namespace Terraria.GameContent.UI.Elements
@@ -95,6 +_,14 @@
 			uIImageButton5.OnMouseOut += ButtonMouseOut;
 			Append(uIImageButton5);
 			num += 24f;
+			uIImageButton5 = new UIImageButton(_buttonRenameTexture);
+			uIImageButton5.VAlign = 1f;
+			uIImageButton5.Left.Set(num, 0f);
+			uIImageButton5.OnClick += CycleDifficulty;
+			uIImageButton5.OnMouseOver += CycleMouseOver;
+			uIImageButton5.OnMouseOut += ButtonMouseOut;
+			Append(uIImageButton5);
+			num += 24f;
 			UIImageButton uIImageButton6 = new UIImageButton(_buttonDeleteTexture) {
 				VAlign = 1f,
 				HAlign = 1f
@@ -270,14 +_,14 @@
 
 			bool flag = Main.ActivePlayerFileData.Player.difficulty == 3;
 			bool isJourneyMode = value.IsJourneyMode;
-			if (flag && !isJourneyMode) {
+			if (flag && !isJourneyMode && !CheatHandler.GetCheat<GamemodeUnlockedWorldCheat>().IsEnabled) {
 				SoundEngine.PlaySound(10);
 				Main.statusText = Language.GetTextValue("UI.PlayerIsCreativeAndWorldIsNotCreative");
 				Main.menuMode = 1000000;
 				return true;
 			}
 
-			if (!flag && isJourneyMode) {
+			if (!flag && isJourneyMode && !CheatHandler.GetCheat<GamemodeUnlockedWorldCheat>().IsEnabled) {
 				SoundEngine.PlaySound(10);
 				Main.statusText = Language.GetTextValue("UI.PlayerIsNotCreativeAndWorldIsCreative");
 				Main.menuMode = 1000000;
@@ -294,6 +_,16 @@
 			uIVirtualKeyboard.SetMaxInputLength(27);
 			Main.MenuUI.SetState(uIVirtualKeyboard);
 			(base.Parent.Parent as UIList)?.UpdateOrder();
+		}
+
+		private void CycleMouseOver(UIMouseEvent evt, UIElement listeningElement) {
+			_buttonLabel.SetText(Language.GetTextValue("UI.CycleWorldDifficulty"));
+		}
+
+		private void CycleDifficulty(UIMouseEvent evt, UIElement listeningElement) {
+			SoundEngine.PlaySound(10);
+			Main.menuMode = 10;
+			//WorldGen.CycleWorldDifficulty(_data);
 		}
 
 		private void OnFinishedSettingName(string name) {
