--- src/Terraria/Terraria/GameContent/Bestiary/CommonEnemyUICollectionInfoProvider.cs
+++ src/Terraclient/Terraria/GameContent/Bestiary/CommonEnemyUICollectionInfoProvider.cs
@@ -1,3 +_,5 @@
+using Terraria.Terraclient;
+using Terraria.Terraclient.Cheats.General;
 using Terraria.UI;
 
 namespace Terraria.GameContent.Bestiary
@@ -20,6 +_,9 @@
 		}
 
 		public static BestiaryEntryUnlockState GetUnlockStateByKillCount(int killCount, bool quickUnlock) {
+			if (CheatHandler.GetCheat<BestiaryForceUnlockCheat>().IsEnabled) 
+				return BestiaryEntryUnlockState.CanShowDropsWithDropRates_4;
+
 			BestiaryEntryUnlockState bestiaryEntryUnlockState = BestiaryEntryUnlockState.NotKnownAtAll_0;
 			if (quickUnlock && killCount > 0)
 				return BestiaryEntryUnlockState.CanShowDropsWithDropRates_4;
