--- src/tModLoader/Terraria/GameContent/Bestiary/GoldCritterUICollectionInfoProvider.cs
+++ src/Terraclient/Terraria/GameContent/Bestiary/GoldCritterUICollectionInfoProvider.cs
@@ -1,4 +_,6 @@
 using Terraria.ID;
+using Terraria.ModLoader;
+using Terraria.Terraclient.Implementation.Defaults;
 
 namespace Terraria.GameContent.Bestiary
 {
@@ -56,7 +_,7 @@
 		}
 
 		private BestiaryEntryUnlockState GetUnlockStateForCritter(string persistentId) {
-			if (!Main.BestiaryTracker.Sights.GetWasNearbyBefore(persistentId))
+			if (!Main.BestiaryTracker.Sights.GetWasNearbyBefore(persistentId) && !ModContent.GetCheatWithFallback<BestiaryForceUnlockCheat>().IsEnabled)
 				return BestiaryEntryUnlockState.NotKnownAtAll_0;
 
 			return BestiaryEntryUnlockState.CanShowDropsWithDropRates_4;
