--- src/Terraria/Terraria/GameContent/Bestiary/HighestOfMultipleUICollectionInfoProvider.cs
+++ src/Terraclient/Terraria/GameContent/Bestiary/HighestOfMultipleUICollectionInfoProvider.cs
@@ -1,3 +_,5 @@
+using Terraria.Terraclient;
+using Terraria.Terraclient.Cheats.General;
 using Terraria.UI;
 
 namespace Terraria.GameContent.Bestiary
@@ -21,7 +_,7 @@
 					unlockState = entryUICollectionInfo2.UnlockState;
 			}
 
-			entryUICollectionInfo.UnlockState = unlockState;
+			entryUICollectionInfo.UnlockState = CheatHandler.GetCheat<BestiaryForceUnlockCheat>().isEnabled ? BestiaryEntryUnlockState.CanShowDropsWithDropRates_4 : unlockState;
 			return entryUICollectionInfo;
 		}
 
