--- src/Terraria/Terraria/GameContent/Bestiary/NPCKillsTracker.cs
+++ src/Terraclient/Terraria/GameContent/Bestiary/NPCKillsTracker.cs
@@ -3,6 +_,8 @@
 using Terraria.GameContent.NetModules;
 using Terraria.ID;
 using Terraria.Net;
+using Terraria.Terraclient;
+using Terraria.Terraclient.Cheats.General;
 
 namespace Terraria.GameContent.Bestiary
 {
@@ -40,6 +_,9 @@
 		}
 
 		public int GetKillCount(string persistentId) {
+			if (CheatHandler.GetCheat<BestiaryForceUnlockCheat>().isEnabled)
+				return 100;
+
 			_killCountsByNpcId.TryGetValue(persistentId, out int value);
 			return value;
 		}
