--- src/tModLoader/Terraria/ModLoader/Mod.cs
+++ src/Terraclient/Terraria/ModLoader/Mod.cs
@@ -362,7 +_,7 @@
 		/// <exception cref="System.Exception">Cannot get packet for " + Name + " because it does not exist on the other side</exception>
 		public ModPacket GetPacket(int capacity = 256) {
 			if (netID < 0)
-				throw new Exception("Cannot get packet for " + Name + " because it does not exist on the other side");
+				Logging.Terraria.Error("Cannot get packet for " + Name + " because it does not exist on the other side");
 
 			var p = new ModPacket(MessageID.ModPacket, capacity + 5);
 			if (ModNet.NetModCount < 256)
