--- src/tModLoader/Terraria/ModLoader/UI/Interface.cs
+++ src/Terraclient/Terraria/ModLoader/UI/Interface.cs
@@ -13,6 +_,7 @@
 using Terraria.ModLoader.UI.ModBrowser;
 using Terraria.GameContent.UI.States;
 using Terraria.Social.Steam;
+using Terraria.Terraclient;
 
 namespace Terraria.ModLoader.UI
 {
@@ -42,6 +_,7 @@
 		internal const int modConfigID = 10024;
 		internal const int createModID = 10025;
 		internal const int exitID = 10026;
+		internal const int TerraclientSettingsID = 10027;
 		internal static UIMods modsMenu = new UIMods();
 		internal static UILoadMods loadMods = new UILoadMods();
 		internal static UIModSources modSources = new UIModSources();
@@ -180,7 +_,7 @@
 			else if (Main.menuMode == tModLoaderSettingsID) {
 				offY = 210;
 				spacing = 42;
-				numButtons = 7;
+				numButtons = 8;
 				buttonVerticalSpacing[numButtons - 1] = 18;
 				for (int i = 0; i < numButtons; i++) {
 					buttonScales[i] = 0.75f;
@@ -252,6 +_,37 @@
 					backButtonDown = false;
 					Main.menuMode = 11;
 					SoundEngine.PlaySound(11, -1, -1, 1);
+				}
+			}
+			else if (Main.menuMode == TerraclientSettingsID) {
+				offY = 210;
+				spacing = 42;
+				numButtons = 3;
+				buttonVerticalSpacing[numButtons - 1] = 18;
+
+				for (int i = 0; i < numButtons; i++) 
+					buttonScales[i] = 0.75f;
+
+				int buttonIndex = 0;
+				buttonNames[buttonIndex] = Client.ConnectToVanillaServers ? Language.GetTextValue("Terraclient.AllowVanillaClients") : Language.GetTextValue("Terraclient.DisallowVanillaClients");
+				if (selectedMenu == buttonIndex) {
+					SoundEngine.PlaySound(SoundID.MenuTick);
+					Client.ConnectToVanillaServers = !Client.ConnectToVanillaServers;
+				}
+
+				buttonIndex++;
+				buttonNames[buttonIndex] = Client.PoseAsVanilla ? Language.GetTextValue("Terraclient.PoseAsVanilla") : Language.GetTextValue("Terraclient.DoNotPoseAsVanilla");
+				if (selectedMenu == buttonIndex) {
+					SoundEngine.PlaySound(SoundID.MenuTick);
+					Client.PoseAsVanilla = !Client.PoseAsVanilla;
+				}
+
+				buttonIndex++;
+				buttonNames[buttonIndex] = Lang.menu[5].Value;
+				if (selectedMenu == buttonIndex || backButtonDown) {
+					backButtonDown = false;
+					Main.menuMode = 11;
+					SoundEngine.PlaySound(11);
 				}
 			}
 			else if (Main.menuMode == modConfigID)
