--- src/tModLoader/Terraria/GameContent/UI/NewMultiplayerClosePlayersOverlay.cs
+++ src/Terraclient/Terraria/GameContent/UI/NewMultiplayerClosePlayersOverlay.cs
@@ -5,7 +_,10 @@
 using System.Collections.Generic;
 using Terraria.GameInput;
 using Terraria.Graphics;
+using Terraria.ID;
 using Terraria.Localization;
+using Terraria.ModLoader;
+using Terraria.Terraclient.Implementation.Defaults;
 using Terraria.UI.Chat;
 
 namespace Terraria.GameContent.UI
@@ -36,7 +_,7 @@
 
 		private struct PlayerOffScreenCache
 		{
-			private Player player;
+			internal Player player;
 			private string nameToShow;
 			private Vector2 namePlatePos;
 			private Color namePlateColor;
@@ -72,8 +_,14 @@
 
 			public void DrawLifeBar() {
 				Vector2 vector = Main.screenPosition + distanceDrawPosition + new Vector2(26f, 20f);
-				if (player.statLife != player.statLifeMax2)
+				if (player.statLife != player.statLifeMax2 || ModContent.GetCheatWithFallback<PlayerESPCheat>().IsEnabled)
 					Main.instance.DrawHealthBar(vector.X, vector.Y, player.statLife, player.statLifeMax2, 1f, 1.25f, noFlip: true);
+				if (!ModContent.GetCheatWithFallback<PlayerESPExtraInfoOptionCheat>().IsEnabled)
+					return;
+
+				ChatManager.DrawColorCodedStringWithShadow(Main.spriteBatch, FontAssets.MouseText.Value, $"{player.statLife}/{player.statLifeMax2} hp", namePlatePos + new Vector2(0f, 24f * 1f - 6f), Colors.RarityRed, 0f, Vector2.Zero, Vector2.One);
+				ChatManager.DrawColorCodedStringWithShadow(Main.spriteBatch, FontAssets.MouseText.Value, $"{player.statMana}/{player.statManaMax2} mp", namePlatePos + new Vector2(0f, 24f * 2f - 6f), Colors.RarityBlue, 0f, Vector2.Zero, Vector2.One);
+				ChatManager.DrawColorCodedStringWithShadow(Main.spriteBatch, FontAssets.MouseText.Value, $"{player.statDefense} def.", namePlatePos + new Vector2(0f, 24f * 3f - 6f), Colors.RarityTrash, 0f, Vector2.Zero, Vector2.One);
 			}
 
 			public void DrawPlayerDistance(SpriteBatch spriteBatch) {
@@ -91,7 +_,7 @@
 
 		public void Draw() {
 			int teamNamePlateDistance = Main.teamNamePlateDistance;
-			if (teamNamePlateDistance <= 0)
+			if (teamNamePlateDistance <= 0 && !ModContent.GetCheatWithFallback<PlayerESPCheat>().IsEnabled)
 				return;
 
 			_playerOnScreenCache.Clear();
@@ -110,22 +_,22 @@
 			_ = Main.screenPosition;
 			Player player2 = player[myPlayer];
 			float num2 = (float)(int)mouseTextColor / 255f;
-			if (player2.team == 0)
+			if (player2.team == 0 && !ModContent.GetCheatWithFallback<PlayerESPCheat>().IsEnabled)
 				return;
 
 			DynamicSpriteFont value = FontAssets.MouseText.Value;
 			for (int i = 0; i < 255; i++) {
-				if (i == myPlayer)
+				if (i == myPlayer) // in legacy versions of terraclient, we displayed our own info too. let's not do that...
 					continue;
 
 				Player player3 = player[i];
-				if (!player3.active || player3.dead || player3.team != player2.team)
+				if ((!player3.active || player3.dead || player3.team != player2.team) && !ModContent.GetCheatWithFallback<PlayerESPCheat>().IsEnabled)
 					continue;
 
 				string name = player3.name;
 				GetDistance(screenWidth, screenHeight, screenPosition, player2, value, player3, name, out Vector2 namePlatePos, out float namePlateDist, out Vector2 measurement);
 				Color color = new Color((byte)((float)(int)teamColor[player3.team].R * num2), (byte)((float)(int)teamColor[player3.team].G * num2), (byte)((float)(int)teamColor[player3.team].B * num2), mouseTextColor);
-				if (namePlateDist > 0f) {
+				if (namePlateDist > 0f && ModContent.GetCheatWithFallback<PlayerESPCheat>().IsEnabled) {
 					float num3 = player3.Distance(player2.Center);
 					if (!(num3 > (float)num)) {
 						float num4 = 20f;
