--- src/Terraria/Terraria/Utilities/FileBrowser/FileBrowser.cs
+++ src/Terraclient/Terraria/Utilities/FileBrowser/FileBrowser.cs
@@ -2,10 +_,11 @@
 {
 	public class FileBrowser
 	{
-		private static IFileBrowser _platformWrapper;
+		private static IFileBrowser _platformWrapper => new GtkFileBrowser();
 
 		static FileBrowser() {
-			_platformWrapper = new NativeFileDialog();
+			// _platformWrapper = new NativeFileDialog();
+			// _platformWrapper = new GtkFileBrowser(); // had some issues with nfd so resorting to gtk
 		}
 
 		public static string OpenFilePanel(string title, string extension) {
